holochain.go

```diff
@@ -612,6 +614,33 @@ func (h *Holochain) Send(proto Protocol, to peer.ID, t MsgType, body interface{}
 	return
 }
 
+func (h *Holochain) Sign(doc []byte) (sig []byte, err error) {
+	privKey := h.agent.PrivKey()
+	sig, err = privKey.Sign(doc)
+	if err != nil {
+		return
+	}
+	return
+}
+
+//TODO
+func (h *Holochain) VerifySignature(signature string, data string, pubKey string) (matches bool, err error) {
+	var pubKeyIC ic.PubKey
+	var sig []byte
+	sig = b58.Decode(signature)
```

Comment:
On thinking about this, I think the b58 encode decode shouldn't be in this function, it should be in the Action.Do() function.  I think signature should be of type []byte and pubKey should be be and actual unmarshaled pubKey, and the decoding is part of the action.